### ISSUE TITLE
Merge webhook resource-spec logic into cloud-api-adaptor

### ISSUE THREAD

**Issue by jtumber-ibm** [confidential-containers/cloud-api-adaptor#1103]
In my opinion the mutating webhook logic does not need to exist as a separate entity and can be included in the cloud-api-adaptor logic. 

As users start using peer pods for high resource workloads, updating the resource spec is a requirement for using peer-pods. So I think it could be worth merging into a single image. But happy if others have reasons I've not thought about to keep it separate.

@bpradipt thoughts?

**Comment by bpradipt** [comment]
Webhook can/should be merged with the CoCo operator like the peerpodconfig and peerpod controllers. The  CAA is implementation for the kata remote hypervisor and it's best to keep these separate imho.
Happy to hear other thoughts.

**Comment by stevenhorsman** [comment]
I think @caavery is working on this as part of #872? So if we update that issues to explicitly state this is covered then we could close this out?

**Comment by bpradipt** [comment]
Changing the title of this issue to mention integrating webhook with CoCo operator. This will be a separate activity from #872

**Comment by caavery** [comment]
Yes I am assuming the webhook is part of #872 and am including it in the work. Sorry this is a bit of a learning process for me.
